client/src/driver/BitBakeProjectScanner.ts

```diff
@@ -412,41 +412,81 @@ You should adjust your docker volumes to use the same URIs as those present on y
+      logger.error('Failed to scan recipes path')
+      return
+    }
 
-    for (const file of tmpFiles) {
-      const recipeName: string = file.name.split(/[_]/g)[0]
+    const output = showRecipeFileNameOutput.output.toString()
+    const splittedOutput = output.split(/\r?\n/g)
+    const startingIndex = splittedOutput.findIndex((line) => line.includes('Available recipes'))
 
-      const element: ElementInfo | undefined = this._bitbakeScanResult._recipes.find((obj: ElementInfo): boolean => {
-        return obj.name === recipeName
-      })
+    const allFileNames: string[] = []
+    if (startingIndex === -1) {
+      logger.error('[scanForRecipesPath] Failed to find available recipes')
+      return
+    }
+
+    const recipePathRegex = /(.*\.bb)/
 
-      if (element !== undefined) {
-        element.path = file.path
+    // All lines after the '=== Available recipes: ===' line are file names, and only keep the valid ones
+    allFileNames.push(...splittedOutput.slice(startingIndex + 1).filter((line) => !line.includes('skipped') && recipePathRegex.exec(line) !== null))
+
+    const notMatchedFileNames: string[] = []
+    for (const fileName of allFileNames) {
+      const match = recipePathRegex.exec(fileName)
+      if (match === null) {
+        continue
       }
-    }
 
-    if (this._shouldDeepExamine) {
-      const recipesWithOutPath: ElementInfo[] = this._bitbakeScanResult._recipes.filter((obj: ElementInfo): boolean => {
-        return obj.path === undefined
+      const recipePath = match[0].trim()
+      const foundRecipe = this._bitbakeScanResult._recipes.find((recipe) => {
+        return recipe.name === extractRecipeName(recipePath)
       })
+      if (foundRecipe !== undefined) {
+        foundRecipe.path = path.parse(recipePath)
+      } else {
+        notMatchedFileNames.push(fileName)
+      }
+    }
 
-      logger.info(`${recipesWithOutPath.length} recipes must be examined more deeply.`)
+    const recipesWithOutPath = this._bitbakeScanResult._recipes.filter((recipe) => {
+      return recipe.path === undefined
+    })
 
-      for (const recipeWithOutPath of recipesWithOutPath) {
-        const commandResult = await this.executeBitBakeCommand(`bitbake-layers show-recipes -f ${recipeWithOutPath.name}`)
-        if (commandResult.status !== 0) {
-          logger.error(`Failed to scan recipes path: ${commandResult.stderr.toString()}`)
-          continue
-        }
-        const output = commandResult.output.toString()
-        const regExp: RegExp = /(\s.*\.bb)/g
+    logger.debug(`${recipesWithOutPath.length} recipes did not get the path from the 'bitbake-layers show-recipes -f' command.`)
+    logger.debug(`${notMatchedFileNames.length} file names (from 'bitbake-layers show-recipes -f) are not matched with any recipes.`)
 
-        for (const match of output.matchAll(regExp)) {
-          recipeWithOutPath.path = path.parse(match[0].trim())
-        }
+    const matchedFileNamesAfterInfer: string[] = []
```

Comment:
Can you split this second search algorithm into a separate function? The function is getting quite big.

Reply:
extracted the deep examine logic and the infer logic into their own functions.